CesiumGS/cesium
# fix out of memory building shaders

# Description

Our `glslToJavascript` build function had some outstanding issues that lead to huge spikes in memory usage when running the local development server. After some performance digging I made a couple small but very impactful changes

- The check for the last modified date for the minifyState cache was _always_ true because the `writeFile` always updated it regardless if it changed. This caused every file to need to be rebuilt. Now it checks first
    - huge reduction in files that need to be processed after the initial run
- The file watcher in the server triggered a full rebuild for all files even if a single one was changed
    - Augmented the `glslToJavascript` function to take specific files to look at. Guarantees it runs super fast on a single file when one changes and no longer bottlenecks
- Pulled out some path variables that do not need to be calculated every loop per file







## Issue number and link

Fixes https://github.com/CesiumGS/cesium/issues/12164





## Testing plan

- Run all the build functions and ensure they still work as expected
- Run `npm start` for the local server
- Modify any shader files and make sure memory does not skyrocket
    - Try this on main and see that it does



# Author checklist
